src/core/Akka/Actor/Address.cs

```diff
@@ -45,7 +45,7 @@ public int Compare(Address x, Address y)
                 if (result != 0) return result;
                 result = string.CompareOrdinal(x.Host ?? "", y.Host ?? "");
                 if (result != 0) return result;
-                result = Nullable.Compare(x.Port, x.Port);
+                result = (x.Port ?? 0).CompareTo(y.Port ?? 0);
```

Comment:
It looks like this might have been source of failing cluster specs. From what I've investigated, `Nullable.Compare<int>(int?, int?)` is not reliable and may return 0 for two different numbers. Waiting for CI to confirm.

Reply:
Nasty. Good find!